src/journey-maps/angular/components/template-renderer/template-renderer.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  TemplateRef,
+  ViewChild,
+} from '@angular/core';
+
+import { SbbTemplateType } from '../../journey-maps.interfaces';
+
+@Component({
+  selector: 'sbb-template-renderer',
+  templateUrl: './template-renderer.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SbbTemplateRenderer implements OnChanges {
+  @Input() templateContext: any;
+  @Input() template?: SbbTemplateType;
+
+  ngTemplate?: TemplateRef<any>;
+  htmlTemplate?: HTMLElement;
+
+  private _htmlTemplateContainer?: ElementRef<HTMLElement>;
+  private _htmlTemplateObserver: MutationObserver;
+
+  @ViewChild('htmlTemplateContainer') set htmlTemplateContainer(
+    container: ElementRef<HTMLElement>
+  ) {
+    this._htmlTemplateContainer = container;
+    this.addHtmlTemplateToDOM();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (!changes.template) {
+      return;
+    }
+
+    this._htmlTemplateObserver?.disconnect();
+
+    if (typeof this.template === 'string') {
+      this.ngTemplate = undefined;
+      const content = (document.getElementById(this.template) as HTMLTemplateElement).content;
+      const setupHtmlTemplate = () => {
+        this.htmlTemplate = content.cloneNode(true) as HTMLElement;
+        this.addHtmlTemplateToDOM();
+      };
+
+      setupHtmlTemplate();
+      this._htmlTemplateObserver = new MutationObserver(setupHtmlTemplate);
+      this._htmlTemplateObserver.observe(content, {
```

Comment:
I believe this would not trigger for changes in attributes or text.
The following should work but is also more expensive.
```ts
const contentObserverConfig: MutationObserverInit = {
  attributes: true,
  childList: true,
  subtree: true,
  characterData: true,
};
```

Reply:
Thanks - I've adopted your suggestion. The templates should not be that big so I think it's ok.